### PR TITLE
[Advanced Compiler]Add-Replication_pad3d

### PR DESCRIPTION
**Pull request by AdvancedCompiler**
### PR Category

Operator
### Type of Change

New Feature
### Description

add Replication_pad3d
### Progress

- [ ] Change is properly reviewed (1 reviewer required, 2 recommended).
- [ ] Change is responded to an issue.
- [ ] Change is fully covered by a UT.

### Performance

f5f76b0e547ce417255a14590b919b72
775b92defabbf120f017a99f3037db73

### CODE REVIEW

File: tests/test_special_ops.py [hunk 1]
```diff
@@ -1899,11 +1899,11 @@ def _verify_expert_level_sorting(
...
+    x = torch.randn(shape, dtype=dtype, device=flag_gems.device)
 
     m_ref = torch.nn.ReplicationPad3d(padding)
-    ref_out = m_ref(x.cpu()).cuda()
+    ref_out = m_ref(x)
     with flag_gems.use_gems():
         res_out_functional = flag_gems.replication_pad3d(x, padding)
 
-    gems_assert_close(res_out_functional, ref_out, dtype, reduce_dim=1)
+    gems_assert_close(res_out_functional.cpu(), ref_out.cpu(), dtype, reduce_dim=1)
```
Comment (tengqm) [inline tests/test_special_ops.py]: why the 'cpu()' call?

Comment (you-and-you) [reply tests/test_special_ops.py]: After ref_out is computed on the GPU, ensure that res_out_functional and ref_out are on the same device.

File: src/flag_gems/__init__.py [hunk 2]
```diff
@@ -298,6 +298,7 @@ def torch_ge(v):
     ("rms_norm", rms_norm),
     ("rsqrt", rsqrt),
     ("rsqrt_", rsqrt_),
+    ("replication_pad3d", replication_pad3d),
```
Comment (tengqm) [inline src/flag_gems/__init__.py]: keep the list sorted, please

File: tests/test_special_ops.py [hunk 3]
```diff
@@ -1890,3 +1890,20 @@ def _verify_expert_level_sorting(
     gems_assert_close(
         num_tokens_post_pad, to_reference(num_tokens_post_pad_vllm), dtype=dtype
     )
+
+
+@pytest.mark.replication_pad3d
+@pytest.mark.parametrize(
+    "shape", [(1, 3, 4, 8, 8), (2, 16, 2, 3, 5), (4, 8, 3, 4, 4), (2, 1, 1, 2, 2)]
```
Comment (tengqm) [inline tests/test_special_ops.py:1936]: Any special consideration about the shapes?
I mean, are these four shapes representing different scenarios?

Comment (you-and-you) [reply tests/test_special_ops.py:1936]: (1, 3, 4, 8, 8) is used as a benchmark; (2, 16, 2, 3, 5) tests multi-channel, non-power-of-two mask computation; (4, 8, 3, 4, 4) is for parallel testing of depth dimension and multi-batch; (2, 1, 1, 2, 2) is an edge case test for single-channel and single-depth.

File: src/flag_gems/ops/replication_pad3d.py [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+    out_offset = offs_h[:, None] * stride_oh + offs_w[None, :] * stride_ow
+
+    mask = (offs_h[:, None] < H_out) & (offs_w[None, :] < W_out)
+
+    vals = tl.load(x_base_ptr + x_offset, mask=mask)
+    tl.store(out_base_ptr + out_offset, vals, mask=mask)
+
+
+def replication_pad3d(x, padding):
+    if isinstance(padding, int):
```
Comment (0x45f) [inline src/flag_gems/ops/replication_pad3d.py:83]: add logger.debug

File: src/flag_gems/ops/replication_pad3d.py [hunk 5]
```diff
@@ -0,0 +1,120 @@
+import torch
+import triton
+import triton.language as tl
+
+
+@triton.autotune(
```
Comment (0x45f) [inline src/flag_gems/ops/replication_pad3d.py]: add tune config in `src/flag_gems/runtime/backend/_nvidia/tune_configs.yaml`

File: src/flag_gems/ops/replication_pad3d.py [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+    configs=[
+        triton.Config({"BLOCK_H": 16, "BLOCK_W": 64}, num_warps=4),
+        triton.Config({"BLOCK_H": 32, "BLOCK_W": 32}, num_warps=4),
+        triton.Config({"BLOCK_H": 64, "BLOCK_W": 16}, num_warps=4),
+        triton.Config({"BLOCK_H": 8, "BLOCK_W": 128}, num_warps=8),
+    ],
+    key=["H_out", "W_out"],
+)
+@triton.jit
+def _ReplicationPad3d_kernel(
```
Comment (0x45f) [inline src/flag_gems/ops/replication_pad3d.py]:
```suggestion
def replication_pad3d_kernel(
```

File: src/flag_gems/ops/replication_pad3d.py [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+
+    out = torch.empty((N, C, D_out, H_out, W_out), device=x.device, dtype=x.dtype)
+
+    grid = lambda META: (
+        triton.cdiv(W_out, META["BLOCK_W"]),
+        triton.cdiv(H_out, META["BLOCK_H"]),
+        N * C * D_out,
+    )
+
+    _ReplicationPad3d_kernel[grid](
```
Comment (0x45f) [inline src/flag_gems/ops/replication_pad3d.py]:
```suggestion
    replication_pad3d_kernel[grid](
```
